scripts/enable-backend.sh

```diff
@@ -0,0 +1,174 @@
+
+source "../setup.env"
+
+# check if BWS_ACCESS_TOKEN is set
+if [[ -z "${BWS_ACCESS_TOKEN}" ]]; then
+  echo "BWS_ACCESS_TOKEN is not set. Abort."
+  exit 1
+fi
+
+# set server-base to EU instance
+bws config server-base https://vault.bitwarden.eu
+
+##############################
+# ask for input data
+##############################
+
+if [ -n "$1" ]; then
+  instance="$1"
+else
+  echo "What is the name of the instance?"
+  read -r instance
+fi
+
+##############################
+# variables
+##############################
+
+path="../../$PREFIX$instance"
+
+# load secrets from Bitwarden Secret Manager
+RENDER_API_CLIENT_ID_DEV=$(bws secret -t "$BWS_ACCESS_TOKEN" get "b53d7a1d-220e-4e07-b1f9-b22700711f79" | jq -r .value)
+RENDER_API_CLIENT_SECRET_DEV=$(bws secret -t "$BWS_ACCESS_TOKEN" get "83a8e38b-fc22-461f-91a0-b22700712b62" | jq -r .value)
+SENTRY_AUTH_TOKEN=$(bws secret -t "$BWS_ACCESS_TOKEN" get "b9a3e1eb-3925-4ed6-93f4-b2270073c82c" | jq -r .value)
+SENTRY_DSN_BACKEND=$(bws secret -t "$BWS_ACCESS_TOKEN" get "a858a580-9643-4330-8667-b2270073d7a6" | jq -r .value)
+
+chars=abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789
+
+isBackendEnabled=0
+isReplicationBackendEnabled=0
+
+# setting backend version. Pinned to prevent config conflicts
+backendVersion=v1.16.0
+
+##############################
+# functions
+##############################
+
+backendEnabledCheck() {
+  if [ ! -f "$path/config/aam-backend-service/application.env" ]; then
+    isBackendEnabled=0
+  else
+    isBackendEnabled=1
+  fi
+}
+
+replicationBackendEnabledCheck() {
+  composeProfiles=$(getVar "$path/.env" COMPOSE_PROFILES)
+
+  if [ "$composeProfiles" == "database-only" ]; then
+    isReplicationBackendEnabled=0
+  else
+    isReplicationBackendEnabled=1
+  fi
+}
+
+setEnv() {
+    local key="$1"
+    local value="$2"
+    local path="$3"
+
+    sed -i "s|^$key=.*|$key=$value|g" "$path" # linux
+    # gsed -i "s|^$key=.*|$key=$value|g" "$path" # macos
+}
+
+# Funktion zum Abrufen der Umgebungsvariablen
+getVar() {
+    local file="$1"
+    local var="$2"
+    local value
+
+    # grep sucht die Zeile mit der Variable, cut extrahiert den Wert
+    value=$(grep "^$var=" "$file" | cut -d '=' -f2-)
+
+    # Falls die Variable nicht existiert oder leer ist, eine Meldung ausgeben
+    if [ -z "$value" ]; then
+      value="n/a"
+    fi
+
+    echo "$value"
+}
+
+generate_password() {
+  password=""
+  for _ in {1..24} ; do
+    password="$password${chars:RANDOM%${#chars}:1}"
+  done
+}
+
+##############################
+# script
+##############################
+
+# check if backend is already enabled for this instance
+backendEnabledCheck
+
+if [ "$isBackendEnabled" == 1 ]; then
+  echo "Backend already enabled for '$instance'.Abort."
+  exit 1
+else
+  echo ""
+fi
+
+replicationBackendEnabledCheck
+
+if [ "$isReplicationBackendEnabled" == 0 ]; then
+  echo "Replication Backend is required for backend. Please enable first. Abort."
```

Comment:
Is that actually true?
I thought, as the replication-backend implements the same endpoints as couchdb, it doesn't matter for other parts of our system if there is a replication-backend or just direct couchdb

Reply:
```suggestion
  # all functionality should be the same with a direct CouchDB without replication-backend. However, some URLs will need to be adapted for this scenario
  echo "Replication Backend is required for backend. Please enable first. Abort."
```